### PR TITLE
 Add WPush Notification Provider 

### PR DESCRIPTION
**Pull request by anhao**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]:
- [x] I have read and understand the pull request rules.

# Description

 Add WPush Notification Provider 

## Type of change

Please delete any options that are not relevant.

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings
- [x] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

image

| Event | Before | After |
|--------|-------|---|
| `UP` | - | paste-image-here |
| `DOWN` | - | paste-image-here |
| Certificate-expiry | - | paste-image-here |
| Testing | - | paste-image-here |

### CODE REVIEW

File: server/notification-providers/wpush.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        }
+    }
+
+    /**
+     * Get the formatted title for message
+     * @param {?object} heartbeatJSON Heartbeat details (For Up/Down only)
+     * @param {?object} monitorJSON Monitor details (For Up/Down only)
+     * @returns {string} Formatted title
+     */
+    checkStatus(heartbeatJSON, monitorJSON) {
```
Comment (CommanderStorm) [inline server/notification-providers/wpush.js:38]: I think `checkStatus` does not really give an accurate meaning what this function does. What do you think about this instead?

```suggestion
    generateTitle(heartbeatJSON, monitorJSON) {
```

File: server/notification-providers/wpush.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+        try {
+            const ret = await axios.post("https://api.wpush.cn/api/v1/send", {
+                "title": this.checkStatus(heartbeatJSON, monitorJSON),
+                "content": msg,
+                "apikey": notification.wpushAPIkey,
+                "channel": notification.wpushChannel
+            });
+            if (ret.data.code !== 0) {
+                return ret.data.message;
```
Comment (chakflying) [inline server/notification-providers/wpush.js:22]: If this is an error condition, maybe we should do `throw new Error(ret.data.message)` instead.

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
Overall, this looks good.
I have left some inline comments below.

I have added a table to the PR-description. Could you fill it with the requisite screnshots?

[Explanation on the second one](https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md):
> Offering notifications is close to the core of what we are as an uptime monitor. Therefore, making sure that they work is also really important. Because testing notification providers is quite time intensive, we mostly offload this onto the person contributing a notification provider.
> 
> To make shure you have tested the notification provider, please include screenshots of the following events in the pull-request description:
> 
> - `UP`/`DOWN`
> - Certificate Expiry via https://expired.badssl.com/
> - Testing (the test button on the notification provider setup page)

File: src/components/notifications/WPush.vue [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<template>
+    <div class="mb-3">
+        <label for="wpush-apikey" class="form-label">WPUSH APIKey</label>
+        <HiddenInput id="wpush-apikey" v-model="$parent.notification.wpushAPIkey" :required="true" autocomplete="new-password" placeholder="WPushxxxxx"></HiddenInput>
+    </div>
+
+    <div class="mb-3">
+        <label for="wpush-channel" class="form-label">发送通道</label>
```
[the remark this answers is not in the document]
Comment (anhao) [reply src/components/notifications/WPush.vue:8]: Think , Detailed channel descriptions can be viewed at the link below

image

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
Please refer to above review. I think you missed this paragraph:

> I have added a table to the PR-description. Could you fill it with the requisite screnshots?
> 
> [Explanation on the second one](https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md):
> > Offering notifications is close to the core of what we are as an uptime monitor. Therefore, making sure that they work is also really important. Because testing notification providers is quite time intensive, we mostly offload this onto the person contributing a notification provider.
> > 
> > To make shure you have tested the notification provider, please include screenshots of the following events in the pull-request description:
> > 
> > - `UP`/`DOWN`
> > - Certificate Expiry via https://expired.badssl.com/
> > - Testing (the test button on the notification provider setup page)

**Comment by CommanderStorm** [comment]
@anhao Thanks for the PR.
Given that you have not responded since May, I am going to close this PR.
We can reopen the PR if you resolve the merge conflicts and provide the requested screenshots as proof that this notification provider works.
Hope you understand 😅

**Comment by anhao** [comment]
@CommanderStorm  
Thanks, I have pushed a new PR  https://github.com/louislam/uptime-kuma/pull/5049

**Pull request closed without merging**